enhancements/machine-api/dedicated-instances.md

```diff
@@ -0,0 +1,184 @@
+
+## Motivation
+
+Some organizations need to make sure that their workloads are not hosted on the same physical hardware as others.
+
+### Goals
+
+- Provide automation similar to what Machine API supports for spot instances. 
+
+### Non-Goals
+
+- TODO
+
+## Proposal
+
+In order to give users the ability to run their workloads on dedicated instances we should do the following things for AWS, GCP and Azure:
+
+- Add ability to enable dedicated instances using Machine's provider spec.
+
+- Validate that provider spec doesn't contain spot instances configuration and dedicated instances at the same time when it's not supported by the cloud provider.
+
+- Make sure that required changes are supported and implemented within the OCP installer (eg. proper control plane creation).
+
+### Implementation Details
+
+For each of the cloud providers that support dedicated instances the implementation will be different.
+
+#### AWS
+
+`Dedicated Instances are Amazon EC2 instances that run in a virtual private cloud (VPC) on hardware that's dedicated to a single customer.`. [AWS Documentation](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/dedicated-instance.html). Each launched instance has a tenancy attribute and it can be configured similar to how we set availability zone.
+
+```
+placement = &ec2.Placement{
+  AvailabilityZone: aws.String(machineProviderConfig.Placement.AvailabilityZone),
+  Tenancy: aws.String(machineProviderConfig.Placement.Tenancy)
+}
+```
+
+That change will require adding `Tenancy` field to provider spec.
+
+```
+type AWSMachineProviderConfig struct {
+  // existing fields
+  ...
+
+  // Placement specifies where to create the instance in AWS
+	Placement Placement `json:"placement"`
+}
+
+// Placement indicates where to create the instance in AWS
+type Placement struct {
+   // existing fields
+   ...
+
+  // Tenancy indicates tenant policy for instance
+  Tenancy string
+}
+
+```
+
+#### Azure
+
+In order to make dedicated VMs work on Azure we need to understand the concept of host groups and hosts.
+[Azure documentation](https://docs.microsoft.com/en-us/azure/virtual-machines/windows/dedicated-hosts).
+
+```
+A host group is a resource that represents a collection of dedicated hosts. You create a host group in a region and an availability zone, and add hosts to it.
+
+A host is a resource, mapped to a physical server in an Azure data center. The physical server is allocated when the host is created. A host is created within a host group. A host has a SKU describing which VM sizes can be created. Each host can host multiple VMs, of different sizes, as long as they are from the same size series.
+
+When creating a VM in Azure, you can select which dedicated host to use for your VM. You have full control as to which VMs are placed on your hosts.
+```
+
+Above means that we are required to have a host group and host created, this should be handled on installer side. 
```

Comment:
This needs to be handle out of band. The installer might/ might not choose to support this.

Reply:
@alexander-demichev how much work do you think this is to do manually? Is this something we can outline here so that it can be fleshed out easily in the docs once this feature is merged?